### PR TITLE
Cybernetic arm implant balance pass (namely: shields)

### PR DESCRIPTION
**Pull request by silicons**
## Why It's Good For The Game

All implants:
Always retracts when EMP'd. I don't know why it was a prob(15) on the strongest EMPs, that's barely anything.
No longer nodrop. Items are instead component hooked to retract if dropped or otherwise removed from the user.

Gun implants:
No longer literally explodes when EMP'd. This is already an adminspawn meme and tasers are nowhere as strong as before. No reason to make them unusable.

Shield implant:
No longer indestructible in that they will need to recharge for 15 seconds after being damaged to the point of breaking. Obviously can't be extended while damaged. Won't knock down the user when ""broken"" temporarily.
Nerfed to 60 integrity from 75. This is 3 laser shots or a bit more than half a shotgun blast.
Rebalanced to be flat 50% block chance instead of 60% projectile and 30% melee. What was the justification for this? Really don't see the need to make them super strong in ranged. 
No longer nodrop. See: all implants. Already damn strong enough with them being able to block lasers and not being able to break.

## Changelog
:cl:
balance: Cybernetic arm implants given a balancing pass. Gun ones don't blow up on EMPs, all of them now retract on EMPs. Shields now work differently and can overload.
tweak: Cybernetic implants are no longer nodrop, instead having a magnetic catch system so they snap back when dropped/removed from the user.
/:cl:

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
The idea was that in melee people are about to hit around the shield, well range wise you have time to react, I didnt want to make them epic always get rather then any of the other types I added/tweaked
I still want the stander riot shield to be the /best/ at melee well other types are better at ranged based combat

File: code/game/objects/items/shields.dm [hunk 1]
```diff
@@ -357,20 +358,47 @@ obj/item/shield/riot/bullet_proof
...
 /obj/item/shield/riot/implant
-	name = "riot tower shield"
-	desc = "A massive shield that can block a lot of attacks and can take a lot of abuse before breaking." //It cant break unless it is removed from the implant
+	name = "telescoping shield implant"
+	desc = "A compact, arm-mounted telescopic shield. While nigh-indestructible when powered by a host user, it will eventually overload from damage. Recharges while inside its implant."
 	item_state = "metal"
 	icon_state = "metal"
-	block_chance = 30 //May be big but hard to move around to block.
+	block_chance = 50
 	slowdown = 1
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/shields.dm:366]: your also dealing with slowdown, when you have it out for that reason too. Its balanced around the "I can block just a bit better ranged well creeping to the baddie" but in melee the antag/graytide can still use toolbox/esword and not get meme clucked by a sec officer with a shield that blocks 50%of melee.

**Comment by Hatterhat** [comment]
> 
> 
> The idea was that in melee people are about to hit around the shield, well range wise you have time to react, I didnt want to make them epic always get rather then any of the other types I added/tweaked
> I still want the stander riot shield to be the /best/ at melee well other types are better at ranged based combat

okay but by giving them a nodrop shield that's harder to break you kind of _did_ give powergaming seccies an "epic always get" shield sort of deal. especially with that spicy 60ranged block. baton still dabs on people in close range.

File: code/game/objects/items/shields.dm (hunk 1 above)
Comment (silicons) [reply code/game/objects/items/shields.dm:366]: Consider getting a real shield then.
This shield doesn't permanently break and blocks lasers. Good damn enough.

**Review by Ghommie — APPROVED** [review]

**Comment by ghost** [comment]
"harder to break" but it's actually impossible to break lol

**Pull request merged**